core/src/avm2/object/script_object.rs

```diff
@@ -189,12 +190,34 @@ impl<'gc> ScriptObjectData<'gc> {
             .map(|cls| cls.inner_class_definition().read().is_sealed())
             .unwrap_or(false)
         {
-            Err(format!(
-                "Cannot get undefined property {:?} on class {:?}",
-                local_name,
-                self.instance_of()
-            )
-            .into())
+            let class_name = if let Some(cls) = self.instance_of() {
+                // Like `to_qualified_name`, but uses a `.` instead of `::` separate
```

Comment:
let's extract this to a function (should this be a QName method?)

Reply:
I'm not sure if this is generally useful - doesn't everything else use the other format?

Reply:
After further thought, I agree. I've extracted this to a method.